command/agent/config_test.go

```diff
@@ -1068,3 +1071,109 @@ func TestReadConfigPaths_dir(t *testing.T) {
 		t.Fatalf("bad: %#v", config)
 	}
 }
+
+func TestUnixSockets(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.SkipNow()
+	}
+
+	usr, err := user.Current()
+	if err != nil {
+		t.Fatal("Could not get current user")
+	}
+
+	tempdir, err := ioutil.TempDir("", "consul-test-")
+	if err != nil {
+		t.Fatal("Could not create a working directory")
+	}
+
+	type SocketTestData struct {
+		Path string
+		Uid  string
+		Gid  string
+		Mode string
+	}
+
+	testUnixSocketPopulation := func(s SocketTestData) (*UnixSocket, error) {
+		return populateUnixSocket("unix://" + s.Path + ";" + s.Uid + ";" + s.Gid + ";" + s.Mode)
+	}
+
+	testUnixSocketPermissions := func(s SocketTestData) error {
+		return adjustUnixSocketPermissions("unix://" + s.Path + ";" + s.Uid + ";" + s.Gid + ";" + s.Mode)
+	}
+
+	_, err = populateUnixSocket("tcp://abc123")
```

Comment:
The other tests in the file use non-scoped `err` and do not combine the check afterwards. I followed existing syntax.